### ISSUE TITLE
[BUG] DiscordRPC Re-Initialization Error

### ISSUE THREAD

**Issue by Thisisauser6443** [VEXSUS/WSA-Discord-RPC#1]
Hi there! Just as the title says, I can't figure out how to enable RPC on the **Release Ready** build. It was working as intended up until 3 days after I started writing out this post.

This is what the unhandled exception log had to offer:

```
DiscordRPC.Exceptions.UninitializedException: Cannot initialize a client that is already initialized
   at DiscordRPC.DiscordRpcClient.Initialize()
   at WSA_Discord_RPC.Menu.EnableToolStripMenuItem_Click(Object sender, EventArgs e)
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStripItem.FireEventInteractive(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStripItem.FireEvent(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

Help would be much appreciated :)))

**Comment by VEXSUS** [comment]
Hello !

I read your bug report. I am sorry this happened to you.

To clarify the crash topic:
In the code exactly **in line 161 to 166 in Menu.vb** there is a function to disable Discord Rich Presence. In line 166 the application restarts itself. I had to work around it because the library that was used to create such Discord RPC did not allow to reinitialize the Discord RPC because it considered that it is already initialized and it is not. Although I still see that it does not quite work correctly.

In 2-3 days I will release an update to a newer version of the application. It will use different library to create such Discord RPC, so please be patient and understanding. I had to rewrite all the code for the last 2 weeks to make better optimization. In addition, I have school so that's why it took 2 weeks. The interface will also be new, in short more pleasant to the eye. You can not see any progress / code editing on github, because for now I do it locally and then publish on github.

So as I wrote above. Please be patient.

Best regards.
* Please do not close your bug report. When I release an update, I will inform you about it here. And we will see if the bug will still occur.

**Comment by Thisisauser6443** [comment]
Ah, that's fine. I'll be waiting :)

**Comment by VEXSUS** [comment]
Hello!

Thank you for your patience and waiting for the new update.

The new update to version 1.4.0 has now been released. Before installing, please read all the information, including those in the "Other Information" category, that pertains to the future of our WSA Discord RPC software.

Link to the new 1.4.0_stable version: https://github.com/VEXSUS/WSA-Discord-RPC/releases/tag/1.4.0_stable

After reinstalling to the latest version, let me know if the problem you described still occurs in the latest version. If the answer is - no, it doesn't occur or something like that, then I will close the ticket and consider the issue resolved.

⚠️ **If you find more vulnerabilities, please don't describe them in the same ticket, you have to create a new ticket for each vulnerability.**

- Best regards.
- R4D1 CEO - Michał Marszał - vexsus

**Comment by Thisisauser6443** [comment]
Hello again :)

Unfortunately, the new update didn't seem to fix the main issue, since RPC doesn't really seem to start (I think), and there's the same unhandled exception error box (Which, I think, is related to RPC not being connected to the Discord app). 

The only circumstance this happens in, is just whenever I launch the program itself, which I just assumed it would be RPC, but it can be bypassed by just hitting _Continue_

Here's the log:
```
System.InvalidCastException: Conversion from string "" to type 'Boolean' is not valid. ---> System.FormatException: Input string was not in a correct format.
   at Microsoft.VisualBasic.CompilerServices.Conversions.ParseDouble(String Value, NumberFormatInfo NumberFormat)
   at Microsoft.VisualBasic.CompilerServices.Conversions.ToBoolean(String Value)
   --- End of inner exception stack trace ---
   at Microsoft.VisualBasic.CompilerServices.Conversions.ToBoolean(String Value)
   at WSA_Discord_RPC.MainMenu.MainMenu_Load(Object sender, EventArgs e)
   at System.EventHandler.Invoke(Object sender, EventArgs e)
   at System.Windows.Forms.Form.OnLoad(EventArgs e)
   at System.Windows.Forms.Form.OnCreateControl()
   at System.Windows.Forms.Control.CreateControl(Boolean fIgnoreVisible)
   at System.Windows.Forms.Control.CreateControl()
   at System.Windows.Forms.Control.WmShowWindow(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   at System.Windows.Forms.ContainerControl.WndProc(Message& m)
   at System.Windows.Forms.Form.WmShowWindow(Message& m)
   at System.Windows.Forms.Form.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
```

**Comment by VEXSUS** [comment]
Hello!

Thank you for the information. I'm glad I was able to help fix this error.
From your unedited message, it appeared that there was another error. I tried installing the program on a virtual machine. After installation, the program started without any problem. I did not have any discord installed on the virtual machine. So I don't know what caused this error. But if it happens to me, I will release a security update that will remove this error.

**I consider the problem solved and close this issue ✅**

Have a great day!
Best regards
R4D1 Software CEO - Michał Marszał - vexsus
